src/_guides/language/concurrency/index.md

```diff
@@ -312,107 +311,144 @@ to implement isolates.
+* Spawns (starts and creates) an isolate
+* Takes a function argument and executes
+* Captures the result
+* Returns the result to the main isolate
+* Terminates the isolate once work is complete
+* Checks, captures, and throws exceptions and errors back to the main isolate
 
-* [`Isolate.spawn()`][] and [`Isolate.exit()`][]
-* [`ReceivePort`][] and [`SendPort`][]
+[`Isolate.run()`]: {{site.dart-api}}/dev/dart-isolate/Isolate/run.html
 
-[`Isolate.exit()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/Isolate/exit.html
-[`Isolate.spawn()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/Isolate/spawn.html
-[`ReceivePort`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/ReceivePort-class.html
-[`SendPort`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/SendPort-class.html
+#### Running an existing method in a new isolate
 
 Here’s the code for the main isolate:
 
 <?code-excerpt "lib/simple_worker_isolate.dart (main)"?>
 ```dart
 void main() async {
   // Read some data.
-  final jsonData = await _parseInBackground();
+  final jsonData = await Isolate.run(_readAndParseJson);
 
-  // Use that data
+  // Use that data.
   print('Number of JSON keys: ${jsonData.length}');
 }
+```
+
+The spawned isolate executes the following code:
 
-// Spawns an isolate and waits for the first message
-Future<Map<String, dynamic>> _parseInBackground() async {
-  final p = ReceivePort();
-  await Isolate.spawn(_readAndParseJson, p.sendPort);
-  return await p.first as Map<String, dynamic>;
+<?code-excerpt "lib/simple_worker_isolate.dart (spawned)"?>
+```dart
+Future<Map<String, dynamic>> _readAndParseJson() async {
+  final fileData = await File(filename).readAsString();
+  final jsonData = jsonDecode(fileData) as Map<String, dynamic>;
+  return jsonData;
 }
 ```
 
-The `_parseInBackground()` function contains the code that
-_spawns_ (creates and starts) the isolate for the background worker,
-and then returns the result:
+1. `Isolate.run()` spawns an isolate for the background worker,
+   while `main()` waits for the result.
 
-1. Before spawning the isolate, the code creates a `ReceivePort`,
-   which enables the worker isolate
-   to send messages to the main isolate.
+2. The new isolate executes the argument passed to `run()`:
+   the function `_readAndParseJson()`.
 
-2. Next is the call to `Isolate.spawn()`,
-   which creates and starts the isolate for the background worker.
-   The first argument to `Isolate.spawn()` is the function that
-   the worker isolate executes: `_readAndParseJson`.
-   The second argument is the `SendPort`
-   that the worker isolate can use to send messages to the main isolate.
-   The code doesn't _create_ a `SendPort`;
-   it uses the `sendPort` property of the `ReceivePort`.
+3. The `return` statement ends the computation,
+   shutting down the worker isolate and
+   sending `jsonData` back to the main isolate as the result.
```

Comment:
**suggestion:** Tighten the language somewhat.
```suggestion
3. The `return` statement shuts down the spawned isolate and
   sends `jsonData` back to the main isolate as the result.
```

Reply:
It's a little pedantic, but `return` doesn't shut down the isolate, Isolate.run() will take this result and call [exit](https://api.dart.dev/dev/2.19.0-388.0.dev/dart-isolate/Isolate/exit.html), which sends the value back to the main isolate.